addons/mail/models/discuss/discuss_channel_member.py

```diff
@@ -212,13 +219,48 @@ def _rtc_join_call(self, check_rtc_session_ids=None):
         self.rtc_session_ids.unlink()
         rtc_session = self.env['discuss.channel.rtc.session'].create({'channel_member_id': self.id})
         current_rtc_sessions, outdated_rtc_sessions = self._rtc_sync_sessions(check_rtc_session_ids=check_rtc_session_ids)
+        if len(current_rtc_sessions) < SFU_MODE_THRESHOLD and self.channel_id.sfu_channel_uuid:
+            self.channel_id.sfu_channel_uuid = None
+        # obtaining a SFU channel
+        sfu_server_url = discuss.get_sfu_url(self.env)
+        key = discuss.get_sfu_key(self.env)
+        if (
+            not self.channel_id.sfu_channel_uuid and sfu_server_url and len(self.channel_id.rtc_session_ids) >= SFU_MODE_THRESHOLD # TODO hard-coded or setting? Increase threshold later
+        ):
+            try:
+                json_web_token = jwt.sign({"iss": f"{self.get_base_url()}:channel:{self.channel_id.id}"}, key=key, ttl=30, algorithm=jwt.ALGORITHM.HS256)
```

Comment:
As this is used in different places, would there be a way to make a safe and simple method that would take the actual params and then sign correctly, and make the request?